app-ios/tutanota/Sources/Crypto/TUTArgon2idFacade.m

```diff
@@ -0,0 +1,20 @@
+#import <Foundation/Foundation.h>
```

Comment:
was there any reason to do it in objc? You can call C directly from Swift

Reply:
All of the interfaces in Crypto (besides the NativeCryptoFacade) were also written in Objective-C, so I assumed it was desired that everything there be written in Objective-C (e.g. some performance reason I don't know much about).

If there is preference to having it in Swift, then I will gladly rewrite it in Swift for you.

I'm familiar with both languages, although I definitely have a much stronger C background than a Swift background.

Reply:
I think it was because other crypto things were already in Obj-C and no one wanted to do a work of rewriting. Plus, OpenSSL API surface is a bit bigger than in this case.

 I think for new code it would be good to use Swift because it's a bit less arcane but if you see that it doesn't work then just leave it as is.

Reply:
OK, I've rewritten the glue in Swift. 6aa105325b044f9d5f1193ccbef33c16cfef7216